src/layered_config_tree/main.py

```diff
@@ -497,7 +526,7 @@ def __setitem__(self, name, value):
             )
         self._set_with_metadata(name, value, layer=None, source=None)
 
-    def __getattr__(self, name):
+    def __getattr__(self, name: str) -> Any:
```

Comment:
This is the one instance I'm unable to move away from Any. I think it should return `Union[ConfigNodeValue, "LayeredConfigTree"]` but this breaks a bunch of the unit tests in a way I'm unable to resolve. The issue seems to be that the tests are doing things like
```
lct = LayeredConfigTree(data, source)
...
lct.test_key.test_key2
```

But when restricting the return (ie, not just Any) then mypy yells b/c various NestedDictValue types don't have a getattr (e.g. str, float, int)

Reply:
I do change setattr in a later commit. I guess this is where my commit-by-commit suggestions gbreaks down since some of this stuff is general cleanup

Reply:
Discussed that the issue is that mpyp doesn't know the returns of a chained attr. Keeping as Any for now